arcgis-ios-sdk-samples/Maps/Create and save a map/SaveAsViewController.swift

```diff
@@ -59,23 +59,14 @@ class SaveAsViewController: UITableViewController {
     
     /// Present the list of folders to choose from.
     func showFolderOptions() {
-        // Set index for the default option.
-        let selectedIndex = selectedFolderIndex ?? portalFolders.count
         // The titles of the portal folders and a "None" option.
-        let folderTitles = portalFolders.map { $0.title ?? "(No Title)" } + ["None"]
+        let folderTitles = portalFolders.map { $0.title ?? "(No Title)" }
         // Prepare the options table view controller and handle selection.
-        let optionsViewController = OptionsTableViewController(labels: folderTitles, selectedIndex: selectedIndex) { [weak self] newIndex in
+        let optionsViewController = OptionsTableViewController(labels: folderTitles, selectedIndex: selectedFolderIndex, allowsEmptySelection: true) { [weak self] newIndex in
             guard let self = self else { return }
-            // If "None" was selected, set selectedFolder to nil.
-            if newIndex == self.portalFolders.count {
-                self.selectedFolderIndex = nil
-            } else {
-                // Select the appropriate folder.
-                self.selectedFolderIndex = newIndex
-            }
+            self.selectedFolderIndex = newIndex
             // Replace the label text with the selected option.
             self.folderLabel.text = self.selectedFolder?.title ?? "None"
-            self.navigationController?.popViewController(animated: true)
```

Comment:
Did you mean to remove this? That's fine; haven't to tap the back button isn't that big of a deal here. Just wanted to double check.

Reply:
Yes I meant to remove that. IMO, automatically popping the view controller seems to make deselection a hassle in this case.

Reply:
If a user wants to select a different folder, they don't have to deselect the current one in order to select the new one, right? So a user could change the selection either by selecting a new row (going from no selection to a selection or changing the existing selection) or by selecting an existing selection (going from a selection to no selection). Either one of those seems like a perfectly fine reason to pop the view controller, since there isn't much else they can do (other than change the selection again, but who would want to do that?). But either way is fine.